### ISSUE TITLE
Next Step Microgrant: Ligo Protocol

### ISSUE THREAD

**Issue by codynhat** [filecoin-project/devgrants#902]
### 1. What is your project? (max 100 words)
The Ligo Protocol is a protocol for car sharing agreements.

It enables peers to discover and transact with each other with minimal trust by defining how agreements are defined, created, and enforced.

### 2. Project links
  
Link to Github repo:
- https://github.com/Ligo-Protocol

Link to demo or website, if applicable:
- https://ligo-book.codyhatfield.me
- https://devpost.com/software/ligo-protocol

  
### 3. How is IPFS, Filecoin, or related technology used in this project? (max 100 words)
[Agreements and State](https://ligo-book.codyhatfield.me/Transacting/Defining%20Agreements/Agreements%20and%20State) will both be stored in IPFS and archived to Filecoin. Our prototype uses Web3.storage and future versions will as well.
  
### 4. How will you improve your project with this grant? What steps will you take to meet this objective? (max 200 words)

Our main objective during this microgrant is to complete a functioning MVP that we can use to legitimize the project and seek future funding. This MVP includes allowing users to create listings, edit listings, and share direct links to listings with potential renters. It also including the initial agreement definitions and enforcement. We have written more details [here](https://hackmd.io/dJoNmvenSHK6S6A2bfWK2g).
 
### 5. Do you agree to share grant reports upon request, including a final grant report at the end of the three month period?

 Yes.

### 6. Does your proposal comply with our Community Code of Conduct?

Yes.

### 7. Links and submissions
We had two independent project submissions in the Chainlink Spring 2022 Hackathon and are joining forces to create a joint MVP.
- https://devpost.com/software/ligo-protocol
- https://devpost.com/software/booketh
  
### Additional questions:
* For each team member(s), please list name, email, Github account, and role in the project.
  * Madhukar Gaida, @madhukar123456, Developer, shroudmg@gmail.com
  * Sahithi Boinpalli, @sahithiboin, Developer, sahithiboin@gmail.com
  * Cody Hatfield, @codynhat, Developer, cody.hatfield@me.com
* How did you learn about our microgrant program?
  * The Filecoin & IPFS Hackathon team reached out to us after completing the hackathon
* If your project was created as part of an event or hackathon:
  * What was the name of the event? (e.g. ETHGlobal NFTHack, Cal Hacks hello:world, Chainlink, CivHacks, GameDevJ, ETHGlobal Scaling Ethereum)
    * Chainlink Spring 2022 Hackathon
  * Please link to your hackathon submission
    * https://devpost.com/software/ligo-protocol

**Comment by ErinOCon** [comment]
Hi @codynhat, thank you for your proposal! Is this project open source?

**Comment by codynhat** [comment]
Yes it is. Is there any specific license it needs to be?

**Comment by ErinOCon** [comment]
Hi @codynhat, thank you for the update. Yes, the license should be MIT, APACHE2, or GPL license for code, or CC-BY-SA 3.0 license for content. Will this be the case for your project?

**Comment by codynhat** [comment]
Yes, it will be MIT

**Comment by codynhat** [comment]
Posting an update on our progress. We have made steady progress on design and implementation of our MVP.
- https://ligo.dev/docs/
- https://github.com/Ligo-Protocol/js-ligo
- https://github.com/Ligo-Protocol/ligo-web-app

Some highlights:
- We have built a prototype that uses Web3.storage to store agreements in IPFS using the DAG-JOSE codec.
- We have been experimenting with different solutions for chat and communication. The latest iteration is using DIDComm over Waku, which uses libp2p. Agreements are also sent in the message itself as an IPLD CAR.
- We have been working on UI development of a prototype marketplace.

This continues to be a side project for all of us, which we hope will change in the future. Our goal is to keep this a public good, and we are exploring various solutions to fund it as a public good.
